libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -51,10 +57,21 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
-    } else {
-      this.store.dispatch(clearSearch());
-    }
+    // get search value from form
+    this.searchForm.valueChanges.pipe(
+        takeUntil(this.unsubscribe$),
```

Comment:
It's preferable to make the `takeUntil` the last operator in the pipe.

Reply:
updated takeUntil as the last operator.